samples/chaincode/cc-tools-demo/README.md

```diff
@@ -0,0 +1,274 @@
+
+Clone the [cc-tools-demo]() repository and copy the [chaincode]() folder. Then paste it in the root directory for cc-tools-demo here.
+
+```bash
+cd ~
+git clone https://github.com/hyperledger-labs/cc-tools-demo.git
+cp -a ~/cc-tools-demo/chaincode/. $FPC_PATH/samples/chaincode/cc-tools-demo/
+cd $FPC_PATH/samples/chaincode/cc-tools-demo
+```
+
+The chaincode code structure is different than normal chaincode as it's using the cc-tools framework
+
+**Note**: remove all test files (`*_test.go`) you find like `txdefs_createNewLibrary_test.go` as they are not part of the chaincode.
+
+## Edit the chaincode to became an FPC chaincode instead of normal fabric
+
+Go to `$FPC_PATH/samples/chaincode/cc-tools-demo/main.go` and create the project structure.
+
+```bash
+cd $FPC_PATH/samples/chaincode/cc-tools-demo
+```
+
+The code presented here is not different from traditional Go Chaincode developed with CC-Tools framework. All the FPC specific protection mechanisms are handled by the FPC framework transparently.
+
+The `main.go` contains the starting point of the chaincode.
+
+To use FPC, we need to add some logic to instantiate our private chaincode and start it.
+To do so, we use `fpc.NewPrivateChaincode(&chaincode.AssetExample{})` and since we're already in the FPC repo, remove the `go.mod` and `go.sum` files so we use the chaincode as part of the FPC package.
+
+CC-tools-demo chaincode has its own packages that are needed, so we run `go get` in the `cc-tools-demo` folder.
+
+Then, add the following code to `$FPC_PATH/samples/chaincode/cc-tools-demo/main.go` in functions `main()` and `runCCaaS()`:
+
+```go
+package main
+
+import (
+	"github.com/hyperledger/fabric-private-chaincode/samples/chaincode/cc-tools-demo/assettypes"
+	"github.com/hyperledger/fabric-private-chaincode/samples/chaincode/cc-tools-demo/datatypes"
+	"github.com/hyperledger/fabric-private-chaincode/samples/chaincode/cc-tools-demo/header"
+
+	fpc "github.com/hyperledger/fabric-private-chaincode/ecc_go/chaincode"
+)
+
+func main() {
+
+	//*CC-Tools Specific code (DO NOT OVERWRITE)*//
+
+	if os.Getenv("RUN_CCAAS") == "true" {
+		err = runCCaaS()
+	} else {
+		if os.Getenv("FPC_ENABLED") == "true" {
+			err = shim.Start(fpc.NewPrivateChaincode(new(CCDemo)))
+		} else {
+			err = shim.Start(new(CCDemo))
+		}
+	}
+
+	//*CC-Tools Specific code (DO NOT OVERWRITE)*//
+
+}
+
+func runCCaaS() error {
+	address := os.Getenv("CHAINCODE_SERVER_ADDRESS")
+	ccid := os.Getenv("CHAINCODE_PKG_ID") //This needs to be replaced
+
+	//*CC-Tools Specific code (DO NOT OVERWRITE)*//
+
+	var cc shim.Chaincode
+	if os.Getenv("FPC_ENABLED") == "true" {
+		cc = fpc.NewPrivateChaincode(new(CCDemo))
+	} else {
+		cc = new(CCDemo)
+	}
+
+
+	server := &shim.ChaincodeServer{
+		CCID:     ccid,
+		Address:  address,
+		CC:       cc,
+		TLSProps: *tlsProps,
+	}
+
+	//*CC-Tools Specific code (DO NOT OVERWRITE)*//
+
+}
+
+```
+
+## Set the needed env vars in the docker-compose file
```

Comment:
I think this step should come after we explained how to build the chaincode container

Reply:
Isn't it better to have it here as these env vars are mentioned in the code?